pyroute2/ext/rawsocket.py

```diff
@@ -55,28 +56,37 @@ class RawSocket(socket):
 
     fprog = None
 
-    def __init__(self, ifname, bpf=None):
-        self.ifname = ifname
+    async def __aexit__(self, *_):
+        self.close()
+
+    async def __aenter__(self):
```

Comment:
Placing any logic into the context manager protocol beside of the manager logic itself means that this object will not be possible to use without context manager syntax. Is it what we want?

Reply:
One can still call `__aenter__` and `__aexit__` manually; and extracting the logic to separate methods adds a tiny bit of complexity while i did not have any use case for it. But if you think it would be better, it's easy to change !